Add verify push sample backend template

robinske opened the pull request:
<!-- Thank you for contributing to the Function Templates project. -->
<!-- Please fill out the template below for your contribution -->

## Description

Adds a sample backend and challenge interface for working with Verify Push code. Still working on the tests but wanted to get this out for review ASAP.

## Checklist

<!-- Before submitting your pull request please make sure you checked the following tasks: -->

- [x] I ran `npm test` locally and it passed without errors.
- [x] I acknowledge that all my contributions will be made under the project's [license](../blob/main/LICENSE).

<!-- To check a task, put a "x" between the brackets, similar to [x] -->

## Related issues

<!-- List any related issues here -->

philnash commented on verify-push-backend/functions/challenge-status.js:
```diff
@@ -0,0 +1,83 @@
+      },
+    });
+    response.setStatusCode(400);
+    return callback(null, response);
```
While this code will work as expected, it may look confusing to someone who doesn't understand how Twilio Functions code is executed, particularly that the callback is the end of the function.

I might suggest something more like this:

```js
function detectMissingParams(paramNames, event) {
  return paramNames.reduce((acc, param) => {
    if (typeof event[param] === "undefined") {
      acc.push(param);
    }
    return acc;
  }, []);
}
```

`detectMissingParams` takes a list of parameter names and the event and returns an array of missing parameters. You can then use it in the body of the function like:

```js
const missingParams = detectMissingParams(["identity", "sid"], event);
if (missingParams.length > 0) {
  response.setStatusCode(400);
  response.setBody({
    error: {
      message: `Missing parameter; please provide: '${missingParams.join(", ")}'.`,
      moreInfo: "https://www.twilio.com/docs/verify/api/challenge#create-a-challenge-resource",
    },
  });
  return callback(null, response);
} 
```

In this case, the error message will also contain all the missing params, not just the first one that was detected missing.

robinske replied:
good call, I like it!

philnash reviewed (changes requested):
The major bodies of the functions look good. I've made some suggestions around detecting missing parameters, some naming and logging, and the hashing as a function. Let me know what you think.

robinske commented on verify-push-backend/assets/index.html:
```diff
@@ -0,0 +1,228 @@
+<!DOCTYPE html>
```
ooo thank you! good to know. I might migrate to that later but this one relies on enough bootstrap that I'm going to keep it as is for now. 

Though do you know if `<!-- APP_INFO_V2 -->` (in the example you linked to) is required in the template somewhere? Or `<!-- EDIT_CODE -->` (the old indicator)?

robinske replied:
Makes sense @anishshandilya - updated to use `<!-- EDIT_CODE_V2 -->` since Verify provides phone numbers for you.

robinske commented on verify-push-backend/assets/index.html:
```diff
@@ -0,0 +1,228 @@
+        identity: $("#identity").val(),
+      });
+
+      fetch("./hash-identity", {
```
AMAZING! TIL. this is why I love your code reviews thank you!

robinske commented:
Updated and added tests @philnash - let me know if you have strong opinions about the template, otherwise I'm going to keep it with bootstrap for now.

philnash reviewed (approved):
I do not have strong feelings about the index.html template. It mostly exists for apps in Quick Deploy that don't otherwise have an interface.

Otherwise, stoked to see this template!
